VmAgent.Core/Interfaces/ProcessRunner.cs

```diff
@@ -84,7 +91,43 @@ public Task CollectLogs(string id, string logsFolder, ISessionHostManager sessio
 
             if (sessionHostManager.VmAgentSettings.EnableCrashDumpProcessing)
             {
-                // TODO
+                try
+                {
+                    string dumpFolder = Path.Combine(logsFolder, VmDirectories.GameDumpsFolderName);
+                    bool dumpFound = false;
+                    try
+                    {
+                        if (!_systemOperations.IsDirectoryEmpty(dumpFolder))
+                        {
+                            dumpFound = true;
+                        }
+                        else
+                        {
+                            // If dumps folder is empty, delete it
+                            _systemOperations.DeleteDirectoryIfExists(dumpFolder);
+                        }
+                    }
+                    catch (DirectoryNotFoundException) { }
+
+                    if (dumpFound)
+                    {
+                        bool shouldDeleteDump = sessionHostManager.SignalDumpFoundAndCheckIfThrottled(id);
+                        if (shouldDeleteDump)
+                        {
+                            _systemOperations.DeleteDirectoryIfExists(dumpFolder);
```

Comment:
the code looks very familiar to the one for containers. I understand we do not have a base class but perhaps we should move this to a common helper class?

Reply:
i ended up creating a base class for them and moved this logic into the ProcessDumps method there